corehq/apps/indicators/utils.py

```diff
@@ -1,10 +1,10 @@
 from couchdbkit import ResourceNotFound
 from django.conf import settings
-from corehq.util.quickcache import quickcache
+from corehq.util.quickcache import skippable_quickcache
 from dimagi.utils.couch.database import get_db
 
 
-@quickcache([], timeout=60)
+@skippable_quickcache([], lambda: settings.UNIT_TESTING, timeout=60)
```

Comment:
Nice find!

Ideally production code should not reference test code. I view the `UNIT_TESTING` flag as an unfortunate artifact of test code that lives in production code. It makes it harder to reason about run-time state when checks for `UNIT_TESTING` are sprinkled throughout the codebase because it means that the code will always behave differently in production than it does in a test environment. To write a test for the cache here you'd need to do something like
```py
with override_settings(UNIT_TESTING=False):
    # test thing that depends on cache and pray that no
    # other parts of the code are conditionally changing
    # their behavior based on the UNIT_TESTING flag
```
What do you think of having `quickcache` disabled by default when running tests, and also provide a decorator/context manager to enable it for tests that test caching behavior? This seems like it would have a few advantages:
- Production code would not reference test code directly. Instead test code would configure production code based on its needs/expectations. Add a setting `ENABLE_QUICKCACHE =  not UNIT_TESTING` (it's unfortunate that we have to use a calculated setting like this, but that will change when we switch to nose and have `testsettings.py` for things like this).
- Tests that depend on `quickcache` will use code (something like `with enable_quickcache():`) to explicitly enable it.
- Tests that do not want caching side effects will not need to worry about clearing caches on setup/teardown.
- Less state leaking between tests.

This might be a big change, so I'm not expecting it to happen immediately.